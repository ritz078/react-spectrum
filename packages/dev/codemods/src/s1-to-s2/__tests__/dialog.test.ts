/*
 * Copyright 2024 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

// @ts-ignore
import {defineSnapshotTest} from 'jscodeshift/dist/testUtils';
import transform from '../src/codemods/codemod';

const test = (name: string, input: string) => {
  defineSnapshotTest(transform, {}, input, name);
};

test('Removes divider', `
import {Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<Dialog>
  <Heading>Test</Heading>
  <Divider />
  <Content>Content</Content>
</Dialog>
`);

test('Moves close function from DialogTrigger to Dialog', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger>
  <Button>Test</Button>
  {(close) => 
    <Dialog>
      <Heading>Test</Heading>
      <Divider />
      <Content>Content</Content>
    </Dialog>
  }
</DialogTrigger>
`);

test('bails when it cannot move the close function', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger>
  <Button>Test</Button>
  {(close) => 
    <ReusableDialog />
  }
</DialogTrigger>
`);

test('Comments out type="tray"', `
import {DialogTrigger, ActionButton, Dialog, Heading, Divider, Content, Text} from '@adobe/react-spectrum';

<DialogTrigger type="tray">
  <ActionButton>Disk Status</ActionButton>
  <Dialog>
    50% disk space remaining.
  </Dialog>
</DialogTrigger>
`);


test('Comments out if type might be "tray"', `
import {DialogTrigger, ActionButton, Dialog, Heading, Divider, Content, Text} from '@adobe/react-spectrum';

<DialogTrigger type={true ? "tray" : "popover"}>
  <ActionButton>Disk Status</ActionButton>
  <Dialog>
    50% disk space remaining.
  </Dialog>
</DialogTrigger>
`);

test('Comments out mobileType', `
import {DialogTrigger, ActionButton, Dialog, Heading, Divider, Content, Text} from '@adobe/react-spectrum';

<>
  <DialogTrigger mobileType="tray">
    <ActionButton>Disk Status</ActionButton>
    <Dialog>
      50% disk space remaining.
    </Dialog>
  </DialogTrigger>
  <DialogTrigger mobileType={true ? "tray" : "fullscreen"}>
    <ActionButton>Disk Status</ActionButton>
    <Dialog>
      50% disk space remaining.
    </Dialog>
  </DialogTrigger>
</>
`);

test('Removes onDismiss and leaves a comment', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger>
  <Button>Test</Button>
  {(close) => 
    <Dialog onDismiss={close}>
      <Heading>Test</Heading>
      <Divider />
      <Content>Content</Content>
    </Dialog>
  }
</DialogTrigger>
`);

test('Moves isDismissable', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger isDismissable>
  <Button>Test</Button>
  <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>
</DialogTrigger>
`);

test('Replaces type="popover" with Popover component', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger type="popover" hideArrow>
  <Button>Test</Button>
  <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>
</DialogTrigger>
`);

test('Replaces type="fullscreen" with FullscreenDialog component', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger type="fullscreen" isKeyboardDismissDisabled>
  <Button>Test</Button>
  <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>
</DialogTrigger>
`);

test('Replaces type="fullscreenTakeover" with FullscreenDialog component', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger type="fullscreenTakeover">
  <Button>Test</Button>
  <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>
</DialogTrigger>
`);

test('Replaces type="fullscreenTakeover" with FullscreenDialog component and close function', `
import {DialogTrigger, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogTrigger type="fullscreenTakeover">
  <Button>Test</Button>
  {(close) => 
    <Dialog>
      <Heading>Test</Heading>
      <Divider />
      <Content>Content</Content>
    </Dialog>
  }
</DialogTrigger>
`);

test('Moves isDismissable from DialogContainer', `
import {DialogContainer, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogContainer isDismissable>
  {showDialog1 && <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>}
  {showDialog2 && <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>}
</DialogContainer>
`);

test('Replaces type="fullscreen" with FullscreenDialog component in DialogContainer', `
import {DialogContainer, Button, Dialog, Heading, Content, Divider} from '@adobe/react-spectrum';

<DialogContainer type="fullscreen" isKeyboardDismissDisabled>
  {showDialog1 && <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>}
  {showDialog2 && <Dialog>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Dialog>}
</DialogContainer>
`);
