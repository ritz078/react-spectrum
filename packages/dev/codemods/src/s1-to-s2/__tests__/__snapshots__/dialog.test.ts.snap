// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments out if type might be "tray" 1`] = `
"import { DialogTrigger, ActionButton, Dialog, Heading, Divider, Content, Text } from "@react-spectrum/s2";

// TODO(S2-upgrade): type="tray" has not been implemented yet.
<DialogTrigger>
  <ActionButton>Disk Status</ActionButton>
  <Dialog>
    50% disk space remaining.
  </Dialog>
</DialogTrigger>"
`;

exports[`Comments out mobileType 1`] = `
"import { DialogTrigger, ActionButton, Dialog, Heading, Divider, Content, Text } from "@react-spectrum/s2";

<>
  // TODO(S2-upgrade): mobileType has not been implemented yet.
  <DialogTrigger>
    <ActionButton>Disk Status</ActionButton>
    <Dialog>
      50% disk space remaining.
    </Dialog>
  </DialogTrigger>
  // TODO(S2-upgrade): mobileType has not been implemented yet.
  <DialogTrigger>
    <ActionButton>Disk Status</ActionButton>
    <Dialog>
      50% disk space remaining.
    </Dialog>
  </DialogTrigger>
</>"
`;

exports[`Comments out type="tray" 1`] = `
"import { DialogTrigger, ActionButton, Dialog, Heading, Divider, Content, Text } from "@react-spectrum/s2";

// TODO(S2-upgrade): type="tray" has not been implemented yet.
<DialogTrigger>
  <ActionButton>Disk Status</ActionButton>
  <Dialog>
    50% disk space remaining.
  </Dialog>
</DialogTrigger>"
`;

exports[`Moves close function from DialogTrigger to Dialog 1`] = `
"import { DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  <Dialog>{(
      {
        close
      }
    ) => <>
      <Heading>Test</Heading>

      <Content>Content</Content>
    </>}</Dialog>
</DialogTrigger>"
`;

exports[`Moves isDismissable 1`] = `
"import { DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  <Dialog isDismissible>
    <Heading>Test</Heading>

    <Content>Content</Content>
  </Dialog>
</DialogTrigger>"
`;

exports[`Moves isDismissable from DialogContainer 1`] = `
"import { DialogContainer, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogContainer>
  {showDialog1 && <Dialog isDismissible>
    <Heading>Test</Heading>

    <Content>Content</Content>
  </Dialog>}
  {showDialog2 && <Dialog isDismissible>
    <Heading>Test</Heading>

    <Content>Content</Content>
  </Dialog>}
</DialogContainer>"
`;

exports[`Removes divider 1`] = `
"import { Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<Dialog>
  <Heading>Test</Heading>

  <Content>Content</Content>
</Dialog>"
`;

exports[`Removes onDismiss and leaves a comment 1`] = `
"import { DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  // onDismiss was removed from Dialog. Use onOpenChange on the DialogTrigger, or onDismiss on the DialogContainer instead
  <Dialog>{(
      {
        close
      }
    ) => <>
      <Heading>Test</Heading>

      <Content>Content</Content>
    </>}</Dialog>
</DialogTrigger>"
`;

exports[`Replaces type="fullscreen" with FullscreenDialog component 1`] = `
"import { FullscreenDialog, DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  <FullscreenDialog isKeyboardDismissDisabled>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </FullscreenDialog>
</DialogTrigger>"
`;

exports[`Replaces type="fullscreen" with FullscreenDialog component in DialogContainer 1`] = `
"import {
  FullscreenDialog,
  DialogContainer,
  Button,
  Dialog,
  Heading,
  Content,
  Divider,
} from "@react-spectrum/s2";

<DialogContainer>
  {showDialog1 && <FullscreenDialog isKeyboardDismissDisabled>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </FullscreenDialog>}
  {showDialog2 && <FullscreenDialog isKeyboardDismissDisabled>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </FullscreenDialog>}
</DialogContainer>"
`;

exports[`Replaces type="fullscreenTakeover" with FullscreenDialog component 1`] = `
"import { FullscreenDialog, DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  <FullscreenDialog variant="fullscreenTakeover">
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </FullscreenDialog>
</DialogTrigger>"
`;

exports[`Replaces type="fullscreenTakeover" with FullscreenDialog component and close function 1`] = `
"import { FullscreenDialog, DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  <FullscreenDialog variant="fullscreenTakeover">{(
      {
        close
      }
    ) => <>
      <Heading>Test</Heading>

      <Content>Content</Content>
    </>}</FullscreenDialog>
</DialogTrigger>"
`;

exports[`Replaces type="popover" with Popover component 1`] = `
"import { Popover, DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  <Popover hideArrow>
    <Heading>Test</Heading>
    <Divider />
    <Content>Content</Content>
  </Popover>
</DialogTrigger>"
`;

exports[`bails when it cannot move the close function 1`] = `
"import { DialogTrigger, Button, Dialog, Heading, Content, Divider } from "@react-spectrum/s2";

<DialogTrigger>
  <Button>Test</Button>
  // TODO(S2-upgrade): update this dialog to move the close function inside
  {(close) => 
    <ReusableDialog />}
</DialogTrigger>"
`;
